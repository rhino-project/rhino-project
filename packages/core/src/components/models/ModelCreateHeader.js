import { isBaseOwned } from 'rhino/utils/models';
import { breadcrumbFor } from 'rhino/utils/ui';
import { useModelCreateContext } from 'rhino/hooks/controllers';
import { useGlobalComponentForModel } from 'rhino/hooks/overrides';
import ModelSection from './ModelSection';

export const ModelCreateHeaderBase = () => {
  const {
    model,
    showParent: { model: parentModel, resource: parent }
  } = useModelCreateContext();

  const breadcrumbs = () => {
    if (!parent) return [];

    if (isBaseOwned(model)) return breadcrumbFor(model, {}, false);

    return breadcrumbFor(parentModel, parent, true);
  };

  return (
    <ModelSection baseClassName="create-header">{breadcrumbs()}</ModelSection>
  );
};

ModelCreateHeaderBase.propTypes = {};

const ModelCreateHeader = (props) =>
  useGlobalComponentForModel('ModelCreateHeader', ModelCreateHeaderBase, props);

export default ModelCreateHeader;
