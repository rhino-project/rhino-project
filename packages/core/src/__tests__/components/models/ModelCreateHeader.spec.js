import ModelCreateHeader from 'rhino/components/models/ModelCreateHeader';
import { sharedModelTests } from './sharedModelTests';

describe('ModelCreateHeader', () => {
  sharedModelTests(ModelCreateHeader);
});
