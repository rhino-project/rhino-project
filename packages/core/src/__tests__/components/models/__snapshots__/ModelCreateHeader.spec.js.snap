// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ModelCreateHeader > should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelCreateHeader > should render with global override shorthand for model 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
